Milestone 2: User interface

PhillipUg opened the pull request:
Updates added to milestone 2 include:
- Prompt the user
- Get user input
- Output the game board interface

PhillipUg committed:
Added main.rb

adamclasic committed:
add main.rb exucutable

adamclasic committed:
update main.rb

PhillipUg committed:
Added get player info method

adamclasic committed:
add loop

PhillipUg committed:
Added initial prompt and input methods

PhillipUg committed:
reverted project set up

PhillipUg committed:
reverted project_setup branch

PhillipUg committed:
Made file executable. Made it output 'hello world'

PhillipUg committed:
Added rubocop and stickler config files. Added gem file

PhillipUg committed:
merged recent changes from project_setup branch

adamclasic committed:
add user interface

adamclasic committed:
update rubocop errors

euqueme reviewed (changes requested):
> ### Pull Request under the review by Maria Eugenia Queme P. (TSE)

### Hello @PhillipUg and @adamclasic  :balloon: :lollipop: 

### This is a good start, :ok_hand:  you've covered almost every behavior in your game's prototype, please address to the next changes :point_down: 

##### Behaviour

* [x] In this step, not everything must work, the point is to make sure that you thought about everything listed below. you might do it only with inline comments. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/logic.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
* [x] Basic game flow is prepared: ([screenshot hint](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/logic.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
  * [x] `[REQUIRED]` game informs player if selected move is invalid
  * [x] `[REQUIRED]` game informs player if selected move is a winning move
  * [x] `[REQUIRED]` game informs player if selected move is a draw move

- It is not necessary you validate these with working code in this branch but there should be at least inline comments where these validations are going to be performed in your game.

- [x] Using out-of-nowhere numbers to terminate a loop, for example, `5.times do ... end` is a signal that something is wrong. Favor the usage of a terminating variable as bellow.

```ruby
game_on = true

# initial instructions for player

while game_on
  # loop for each move
  
  if winner # and/or draw (the exact condition in this milestone is not important)
    game_on = false
  end
end
```
[OPTIONAL]

- [x] I see you use the sleep method, I strongly advise to find a way to block the keyboard while this method executes or remove it from your project

### That's it! :raised_hands: 

### Please :pray: do the changes :technologist: :technologist: and submit back for a code review when you are done :memo: 

Happy coding! :rocket: 

Any questions? ask me on slack **@ Maru** :woman_technologist: 

PROJECT STATUS: REQUESTED CHANGES :vertical_traffic_light:

PhillipUg commented:
Thanks @euqueme for this detailed review. We shall effect the said changes and submit another review

PhillipUg commented:
## Corrections made to previous PR:
- added initial instructions for player
- added terminating variables to check status of game
- removed `sleep()` method
- removed `5.times loop...end` and instead used terminating variables as suggested
- added inline comments to explain our mock code

whiz25 reviewed (changes requested):
## Changes Required
I don't see any changes you have made regarding the previous reviewer's suggestions.
- [x] Your code still uses `10.times do loop...` instead of variables to terminate the loop. It's recommended to use a while loop.
- [x] How will you handle the win/draw situations? Please let us know

### Martin Afani

PhillipUg committed:
added starting instructions and terminating variables

PhillipUg committed:
Corrected errors flagged by rubocop

PhillipUg commented:
Sorry we mistakenly sent updates to the forked repo instead. Mistake corrected

PhillipUg committed:
Corrected rubocop errors

PhillipUg committed:
corrected rubocop errors

adamclasic committed:
update rubocop.yml

suretrust commented on bin/main.rb:
```diff
@@ -0,0 +1,72 @@
+
+  winner = true
+  game_on = false if winner # terminates when we have a winner
+
```
It's awesome that the game terminates when there is a winner, but we must also inform the players that there is a winner. You can check this from the former reviewer's suggestions.

- [x]  `[REQUIRED]` game informs the players if the selected move is a winning move

PhillipUg replied:
Thanks for catching that. making another request

suretrust reviewed (changes requested):
Hi team, 👋 

### Great job implementing some of the suggested changes. 🙌 Just one more observation, please see my comment below. 👇 

**Kindly submit for another code review request once this has been fixed.** _Happy coding!_ 🎉

PhillipUg committed:
Added output message for winner of game

PhillipUg committed:
corrected rubocop errors

PhillipUg commented:
## Corrected:
- added output message to inform player if they win the game

stratospherique reviewed (approved):
### Great work implementing the suggested changes :clap: :clap: Congrats, the milestone is approved :heavy_check_mark: You may merge :+1: :+1: :+1: 

Keep up the good work :muscle: 
**_Ahmed Mahfoudh_**
